packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -198,113 +197,59 @@ def _get_field(mcs, cls: Type, field_name: str) -> Any:
         return getattr(cls, field_name)
 
 
+@dataclass(frozen=True)
 class BaseTxPayload(ABC, metaclass=_MetaPayload):
     """This class represents a base class for transaction payload classes."""
 
-    transaction_type: Any
-
-    def __init__(
-        self,
-        sender: str,
-        id_: Optional[str] = None,
-        round_count: int = ROUND_COUNT_DEFAULT,
-        **kwargs: Any,
-    ) -> None:
-        """
-        Initialize a transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param id_: the id of the transaction
-        :param round_count: the count of the round in which the payload was sent
-        :param kwargs: the keyword arguments
-        """
-        self.id_ = uuid.uuid4().hex if id_ is None else id_
-        self._round_count = round_count
-        self._sender = sender
-        # kwargs exists for typing only
-        enforce(len(kwargs) == 0, f"kwargs passed to {type(self)} must be empty")
+    sender: str
+    transaction_type: Any = field(init=False)
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
+    id_: str = field(default_factory=lambda: uuid.uuid4().hex, init=False)
 
     @property
-    def sender(self) -> str:
-        """Get the sender."""
-        return self._sender
+    def data(self) -> Dict[str, Any]:
+        """Data"""
+        excluded = ["sender", "round_count", "id_", "transaction_type"]
+        return {k: v for k, v in asdict(self).items() if k not in excluded}
```

Comment:
`content` might be a better nomer. It concerns everything that is relevant for use in the application. ignoring `id_` and `transaction_type` as they will be removed, both `sender` and `round_count` are only relevant for payload processing (metadata so you will); it is not part of what the agents need to reach consensus on.